dmscripts/publish_draft_services.py

```diff
@@ -158,12 +156,14 @@ def publish_draft_service(
                     draft_service['id'],
                     draft_service['serviceId'],
                 )
-                return service_id, True
+                previously_published = True
             else:
                 raise e
 
+    return service_id, previously_published
 
-def _get_draft_services_iter(client, framework_slug, draft_ids_file=None):
+
+def get_draft_services_iter(client, framework_slug, draft_ids_file=None):
     supplier_frameworks = find_suppliers_on_framework(client, framework_slug)
```

Comment:
Why do we need to get `supplier_frameworks`? Couldn't we just use `find_draft_services_by_framework_iter` without a supplier id?

Reply:
We need to filter out any suppliers that failed the framework.

Reply:
Right got it thanks